string goal
---
int16 error
